app/models/location.rb

```diff
@@ -1,12 +1,13 @@
 class Location < ApplicationRecord
-  belongs_to :service_area
 
   has_many :community_resources
   has_many :listings
   has_many :matches
   has_many :organizations
   has_many :people
 
+  validates :state, presence: true
+
```

Comment:
@exbinary yeah, exactly -- we're on same page about avoiding creating empty records

this validation is such that IF we create a `Location,` it needs `:state` at minimum. so, someone could fill out the form without adding state (and maybe we should consider storing state as a default system_setting?) and we would get a Listing that belongs to a ServiceArea, but no Location. 

Do you think we should allow people to fill out Forms with only partial address data (e.g. cross street only)?

In general my preference is to have minimum constraints at both the model and db layers to avoid empty records. We don't have any db constraints in the db `locations` table, and since we're still flexing this usage in anticipation of mapping (maybe we only have lat lng at that point...), it seems reasonable to leave it that way for now.

Reply:
If it helps any, I can definitely think service areas that we might want to span a state border. The MD border of DC's Takoma has some places where walking across the border doesn't feel much like leaving the neighborhood, more so with Delmar on the DE/MD line where even the school district with its one high school is on both sides of the border

Reply:
Totally re service areas spanning states. Just wanting to require one field on Location so we don't create empty location records. Rn there's no connection between service areas and location (we removed it).